lib/hobo_fields/model/field_spec.rb

```diff
@@ -2,15 +2,35 @@
 
 module HoboFields
   module Model
-
     class FieldSpec
+      class UnknownSqlTypeError < RuntimeError; end
 
-      UTF8_BYTES_PER_CHAR = 3
+      MYSQL_TINYTEXT_LIMIT    =        0xff
+      MYSQL_TEXT_LIMIT        =      0xffff
+      MYSQL_MEDIUMTEXT_LIMIT  =   0xff_ffff
+      MYSQL_LONGTEXT_LIMIT    = 0xffff_ffff
+
+      class << self
+        # method for easy stubbing in tests
+        def mysql_text_limits?
+          if defined?(@mysql_text_limits)
+            @mysql_text_limits
+          else
+            @mysql_text_limits = ActiveRecord::Base.connection.class.name.match?(/mysql/i)
+          end
+        end
 
-      class UnknownSqlTypeError < RuntimeError; end
+        def round_up_mysql_text_limit(limit)
+          [MYSQL_TINYTEXT_LIMIT, MYSQL_TEXT_LIMIT, MYSQL_MEDIUMTEXT_LIMIT].find do |mysql_supported_text_limit|
+            if limit <= mysql_supported_text_limit
+              mysql_supported_text_limit
+            end
+          end || MYSQL_LONGTEXT_LIMIT
```

Comment:
Should we raise an error if the limit is trying to be set higher than `MYSQL_LONGTEXT_LIMIT`? I think this would be a rather odd side-effect to snap down in that case without any kind of warning or error.

Reply:
Good idea. It was bugging me that it wasn't symmetrical. Now it will be.

Reply:
Changed here: https://github.com/Invoca/hobo_fields/pull/25/commits/3dc4c61f6535da2c3b9eeb50a81caaa9e177840d
Note that the exception is raised when the `fields` block is run, vs. when the migration is run. I suppose that could be surprising if you generated migrations using non-MySQL and then ran on MySQL? That seems pretty unlikely that you'd do that _and_ mess up the `limit:` by specifying a monstrous one like that.